src/coord/src/coord.rs

```diff
@@ -544,6 +550,28 @@ impl Coordinator {
         }
         // These can occasionally be equal, so ignore the update in that case.
         if next_ts > self.closed_up_to {
+            // Close out the timestamp for persisted tables.
+            //
+            // TODO: Allow sealing multiple streams at once to reduce the
+            // overhead.
+            let (tx, rx) = std::sync::mpsc::channel();
+            for entry in self.catalog.entries() {
+                if let CatalogItem::Table(Table {
+                    persist: Some(persist),
+                    ..
+                }) = entry.item()
+                {
+                    persist.write_handle.seal(next_ts, tx.clone().into());
```

Comment:
This might address my concern about table compaction, and this thing here makes sure persistence writes are complete before closing the ts?

Reply:
The write call and this call both enqueue something in a channel before returning, similar to how SequencedCommand works. The channel is then processed in order, so as long as the calls are serialized, I _think_ this is okay?